plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -3051,6 +3056,23 @@ void producer_plugin::received_block(uint32_t block_num, chain::fork_db_add_t fo
    }
 }
 
+// thread-safe, called when ctrl-c/SIGINT/SIGTERM/SIGPIPE is received
+void producer_plugin::interrupt() {
+   fc_ilog(_log, "interrupt");
+   my->interrupt_read_only();
+   my->chain_plug->chain().interrupt_apply_block_transaction();
+}
+
+void producer_plugin_impl::interrupt_read_only() {
+   if (!_ro_timers.empty() && _ro_max_trx_time_us > fc::milliseconds(250)) {
```

Comment:
Why not allow interrupt for any value of `_ro_max_trx_time_us`, if the user wants to do a ctrl-C?

Reply:
My thought was that if a read-only trx already has a limit of <= 250ms then it will finish in at most 250ms which is not very long so no reason to kill it, just let it finish.